v0/destinations/one_signal/transform.js

```diff
@@ -0,0 +1,291 @@
+        properties[item.propertyName] ||
+        properties[item.propertyName] === ""
+      ) {
+        if (eventAsTags && typeof properties[item.propertyName] === "string") {
+          tags[`${event}_${[item.propertyName]}`] =
+            properties[item.propertyName];
+        } else if (typeof properties[item.propertyName] === "string") {
+          tags[item.propertyName] = properties[item.propertyName];
+        }
+      }
+    });
+  }
+  payload.tags = tags;
+  return responseBuilder(payload, endpoint, message.type);
+};
+
+/**
+ * This function is used to build the response for group call.
+ * @param {*} message
+ * @param {*} param1
+ * @returns
+ */
+const groupResponseBuilder = (message, { Config }) => {
+  const { appId, allowedProperties } = Config;
+  const groupId = getFieldValueFromMessage(message, "groupId");
+  if (!groupId) {
+    throw new CustomError(
+      "[OneSignal]: groupId is required for group events",
+      400
+    );
+  }
+  let { endpoint } = ENDPOINTS.GROUP;
+  const externalUserId = getFieldValueFromMessage(message, "userIdOnly");
+
+  if (!externalUserId) {
+    throw new CustomError(
+      "[OneSignal]: userId is required for group events",
+      400
+    );
+  }
+  endpoint = `${endpoint}/${appId}/users/${externalUserId}`;
+  const payload = {};
+  const tags = {};
+  tags.groupId = groupId;
+
+  // Populating tags using allowed properties(from dashboard)
+  const properties = getFieldValueFromMessage(message, "traits");
+  if (properties && allowedProperties && Array.isArray(allowedProperties)) {
+    allowedProperties.forEach(item => {
+      if (
+        properties[item.propertyName] ||
+        properties[item.propertyName] === ""
+      ) {
+        if (typeof properties[item.propertyName] === "string") {
+          tags[item.propertyName] = properties[item.propertyName];
+        }
+      }
+    });
+  }
+  payload.tags = tags;
+  return responseBuilder(payload, endpoint, message.type);
+};
+
+const processEvent = (message, destination) => {
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+  if (!destination.Config.appId) {
+    throw new CustomError("[OneSignal]: appId is a required field", 400);
+  }
+  const messageType = message.type.toLowerCase();
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = identifyResponseBuilder(message, destination);
+      break;
+    case EventType.TRACK:
+      response = trackResponseBuilder(message, destination);
+      break;
+    case EventType.GROUP:
+      response = groupResponseBuilder(message, destination);
+      break;
+    default:
+      throw new CustomError(`Message type ${messageType} not supported`, 400);
+  }
+  return response;
+};
+
+const process = event => {
+  return processEvent(event.message, event.destination);
+};
+
+const processRouterDest = inputs => {
+  if (!Array.isArray(inputs) || inputs.length <= 0) {
```

Comment:
inputs.length can be -ve?

Reply:
We never would get this scenario, I think they added this case as a guard block